### PR TITLE
[AIRFLOW-2421] - HTTPHook set verify True by default 

### PR DESCRIPTION
**Pull request by OmerJog**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-2421) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2421
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).

### Description

change the hook to use secure connection by default.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Review by Fokko — APPROVED** [review]
I think this is a good change for Airflow 2.0. I prefer to be secure by default.

**Comment by OmerJog** [comment]
I believe this can also be in 1.10.3
While this changes the default behavior it's very easy to restore the previous non secure behavior and it's also shown how on the updaing.md.
This is not a ground braking change.

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,11 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### HTTPHook verify default value changed from False to True.
+
+The HTTPHook is now secured by default: verify=True.
+This can be overwriten by using the extra_options param as {'verify': False}.
```
Comment (mik-laj) [inline UPDATING.md]:
```suggestion
This can be overwriten by using the extra_options param as `{'verify': False}`.
```

File: UPDATING.md [hunk 2]
```diff
@@ -24,6 +24,11 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### HTTPHook verify default value changed from False to True.
+
+The HTTPHook is now secured by default: verify=True.
```
Comment (mik-laj) [inline UPDATING.md]:
```suggestion
The HTTPHook is now secured by default: `verify=True`.
```

Comment (mik-laj) [inline UPDATING.md]: I think it's worth adding a link to the class description. Now it is not possible in a clean way, but when my change is accepted, we should do it and add a note under the parameter description also.
Reference: 
https://github.com/apache/airflow/pull/4827/files

Comment (OmerJog) [reply UPDATING.md]: Seems a bit excessive isn't it? Didn't see any other project that in it's change log adds links to each change class. Not to forget the people who reads this change log are programmers who are familiar with Airflow.

Comment (mik-laj) [reply UPDATING.md]: I now have open documentation for boto3. This contains links to classes documentation.  
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/upgrading.html
Screenshot 2019-03-11 at 14 35 49
It is not difficult to add linking. All you need to do is enter the full name of the class.
Example:
Screenshot 2019-03-11 at 14 37 39

Comment (mik-laj) [reply UPDATING.md]: I do not want you to add a link now. I only think about plans for future changes and improvement. Now it is inadvisable and difficult to do.

Comment (OmerJog) [reply UPDATING.md]: OK. So if anything else is needed let me know.
Currently there are no more comments so waiting for merge.

**Comment by Fokko** [comment]
@OmerJog Can you rebase? I'll merge for Airflow 2.0

**Comment by OmerJog** [comment]
@Fokko  Done

**Comment by OmerJog** [comment]
@Fokko can you take a look ?

**Comment by Fokko** [comment]
I've restarted the CI

**Comment by OmerJog** [comment]
@Fokko please let me know when you have time to review this again so I won't have to rebase for no reason.  Thanks :)

**Pull request merged**